cli_tools/gce_ovf_import/ovf_utils/ovf_utils.go

```diff
@@ -35,42 +35,169 @@ const (
+	44:  {description: "GNU Hurd", importerOSIDs: []string{}},
+	45:  {description: "OS9", importerOSIDs: []string{}},
+	46:  {description: "MACH Kernel", importerOSIDs: []string{}},
+	47:  {description: "Inferno", importerOSIDs: []string{}},
+	48:  {description: "QNX", importerOSIDs: []string{}},
+	49:  {description: "EPOC", importerOSIDs: []string{}},
+	50:  {description: "IxWorks", importerOSIDs: []string{}},
+	51:  {description: "VxWorks", importerOSIDs: []string{}},
+	52:  {description: "MiNT", importerOSIDs: []string{}},
+	53:  {description: "BeOS", importerOSIDs: []string{}},
+	54:  {description: "HP MPE", importerOSIDs: []string{}},
+	55:  {description: "NextStep", importerOSIDs: []string{}},
+	56:  {description: "PalmPilot", importerOSIDs: []string{}},
+	57:  {description: "Rhapsody", importerOSIDs: []string{}},
+	58:  {description: "Windows 2000", importerOSIDs: []string{}},
+	59:  {description: "Dedicated", importerOSIDs: []string{}},
+	60:  {description: "OS/390", importerOSIDs: []string{}},
+	61:  {description: "VSE", importerOSIDs: []string{}},
+	62:  {description: "TPF", importerOSIDs: []string{}},
+	63:  {description: "Windows (R) Me", importerOSIDs: []string{}},
+	64:  {description: "Caldera Open UNIX", importerOSIDs: []string{}},
+	65:  {description: "OpenBSD", importerOSIDs: []string{}},
+	66:  {description: "Not Applicable", importerOSIDs: []string{}},
+	67:  {description: "Windows XP", importerOSIDs: []string{}},
+	68:  {description: "z/OS", importerOSIDs: []string{}},
+	69:  {description: "Microsoft Windows Server 2003", importerOSIDs: []string{}},
+	70:  {description: "Microsoft Windows Server 2003 64-Bit", importerOSIDs: []string{}},
+	71:  {description: "Windows XP 64-Bit", importerOSIDs: []string{}},
+	72:  {description: "Windows XP Embedded", importerOSIDs: []string{}},
+	73:  {description: "Windows Vista", importerOSIDs: []string{}},
+	74:  {description: "Windows Vista 64-Bit", importerOSIDs: []string{}},
+	75:  {description: "Windows Embedded for Point of Service", importerOSIDs: []string{}},
+	76:  {description: "Microsoft Windows Server 2008", importerOSIDs: []string{}},
+	77:  {description: "Microsoft Windows Server 2008 64-Bit", importerOSIDs: []string{}},
+	78:  {description: "FreeBSD 64-Bit", importerOSIDs: []string{}},
+	79:  {description: "RedHat Enterprise Linux", importerOSIDs: []string{}},
+	80:  {description: "RedHat Enterprise Linux 64-Bit", importerOSIDs: []string{"rhel-6", "rhel-6-byol", "rhel-7", "rhel-7-byol", "rhel-8", "rhel-8-byol"}},
+	81:  {description: "Solaris 64-Bit", importerOSIDs: []string{}},
+	82:  {description: "SUSE", importerOSIDs: []string{}},
+	83:  {description: "SUSE 64-Bit", importerOSIDs: []string{"opensuse-15"}, nonDeterministic: true},
+	84:  {description: "SLES", importerOSIDs: []string{}},
+	85:  {description: "SLES 64-Bit", importerOSIDs: []string{"sles-12-byol", "sles-15-byol"}},
+	86:  {description: "Novell OES", importerOSIDs: []string{}},
```

Comment:
As I read through this map, it appears that all entries are empty, marked as nonDetermistic, or have multiple possible matches. From my subsequent reading of the code, it looks like all such candidates wouldn't be used to allow the import to proceed. For example, if an OVF only specified id=83, then the import would stop and prompt the user to specify their `--os`.

Is my understanding correct? If so, will supporting osIDs allow imports to proceed that would have otherwise failed?

Reply:
Looking at the mapping, you're correct. The benefit to the customer is that, instead of throwing a "no mapping found" error, we will throw an error saying we know what is in the OVF descriptor but that it can't be mapped in a 1:1 manner to OS values GCE recognizes. 

For example, when ID=96 was provided, if the description didn't match VMWare specs (e.g. if OVF came from vBox), we would throw an error saying OS is not recognized. With this change, we'll instruct the user to provide either `debian-8` or `debian-9`.

Reply:
I see -- yeah, I agree, the new error messages that are introduced by this PR are a lot more helpful.